app/src/main/java/org/greenstand/android/TreeTracker/managers/LocationUpdateManager.kt

```diff
@@ -120,3 +129,51 @@ fun Location?.accuracyStatus(): Accuracy {
         Accuracy.BAD
     }
 }
+
+class LocationDataCapturer(
+    private val userManager: UserManager,
+    private val locationUpdateManager: LocationUpdateManager,
+    private val treeTrackerDAO: TreeTrackerDAO
+) : KoinComponent {
```

Comment:
Why is this a KoinComponent?

Reply:
It allows to use the `get()` making an eager instantiation possible.

Reply:
This ties our code to Koin, so I think it's best to avoid it. If the only reason is eager instantiation, then we probably don't need it. Ideally everything should be lazy until we need it.